datacommons/utils.py

```diff
@@ -48,84 +50,49 @@
 _MAX_LIMIT = 100
 
 
-# ------------------------ SELECT AND PROCESS HELPERS -------------------------
+# ------------------------- PANDAS UTILITY FUNCTIONS --------------------------
 
 
-def convert_type(col_names, dtype):
-  """ Converts values in a given column to the given type.
+def flatten_frame(pd_frame):
+  """ Expands each cell in a Pandas DataFrame containing a list of values.
 
   Args:
-    col_names: The column or columns to convert
-    dtype: Data type or a dictionary from column name to data type.
-
-  Returns: A process function that converts the column to a given type.
+    pd_frame: The Pandas DataFrame.
   """
-  if isinstance(col_names, str):
-    col_names = [col_names]
-  def process(pd_frame):
-    for name in col_names:
-      pd_frame[name] = pd.to_numeric(pd_frame[name])
-    return pd_frame
-  return process
-
-def drop_nan(col_names):
-  """ Drops rows containing NAN as a value in columns in col_names.
+  for col in pd_frame:
+    if any(isinstance(v, list) for v in pd_frame[col]):
+      # TODO: Uncomment after colab supports pandas 0.25
+      # pd_frame = pd_frame.explode(col)
+      pd_frame = explode(pd_frame, col)
+  pd_frame = pd_frame.reset_index(drop=True)
+  return pd_frame
 
-  Args:
-    col_names: single column name or a list of column names.
-  """
-  if isinstance(col_names, str):
-    col_names = [col_names]
-  def process(pd_frame):
-    return pd_frame.dropna(subset=col_names)
-  return process
+def clean_frame(pd_frame):
+  """ A convenience function that cleans a pandas DataFrame.
 
-def delete_column(*cols):
-  """ Returns a function that deletes the given column from a frame.
+  The following operations are performed:
+  - Columns containing numerical types are converted to floats.
+  - Rows with empty values are dropped.
```

Comment:
We might not want to do this but leave it to user. As this has side effects of removing arbitrary rows..

Reply:
Discussed offline, decided to keep for now.